pkg/docker/docker.go

```diff
@@ -800,15 +926,18 @@ func (d *stiDocker) RunContainer(opts RunContainerOptions) error {
 			// the container exits normally).  dump port/etc information for the user.
 			dumpContainerInfo(container, d, image)
 		}
+
 		// Return an error if the exit code of the container is
 		// non-zero.
 		glog.V(4).Infof("Waiting for container %q to stop ...", containerName)
-		exitCode, err := d.client.WaitContainer(container.ID)
+		ctx, cancel := getDefaultContext(15 * time.Minute)
```

Comment:
a combination of things:
1) the k8s client talks about short term ops and long term ops (pull/build image, attach/wait container) ... the default in k8s for short term is 2 minutes
2) the go-dockerclient did not appear to set a timeout on the client side by default, they just allow you to set them (we never did, but used util.TimeoutAfter around some operations, where are default was 20 seconds);  so go-dockerclient's default is 0; following the trail down into go's http support and time support (see go http's `setRequestCancel` in client.go), it was blocking forever (essentially the now time - 0)

That correlates with what I remember from adding the `--run` support.  If the container stays up, you had to ctrl-c to finish the `s2i` command.

And I believe I've heard of builds "hanging" before (correct me if I'm wrong).

Background done.  Rationale:
1) I was uncomfortable waiting forever on both, though perhaps it makes sense for the run container ... I could put the max int/log there 
2) for the build, 15 was based on some experimentation and ad hoc heuristics ... I'm certainly amenable to suggestions

Bottom line, waiting forever